src/wng_building/station_river_creator.py

```diff
@@ -1,3 +1,124 @@
+        stations is a dictionary with reg-number keys and dictionary values like
+        2275: {
+            'river_name': Tisza,
+            'station_name': 'Szeged',
+            'EOVy': 735218.1,
+            'EOVx': 101317.2,
+            'null_point': 73.70
+        }
+        :return dict: dictionary of all stations
+        """
+        stations = {}
+        for reg_num in list(self.data_handler.reg_station_mapping.keys()):
+            station_name = self.data_handler.reg_station_mapping[reg_num]
+            river_name = self.data_handler.station_river_mapping[station_name]
+
+            stations[reg_num] = {
+                'river_name': river_name,
+                'station_name': station_name,
+                'EOVy': self.data_handler.station_coordinates[reg_num]['EOVy'],
+                'EOVx': self.data_handler.station_coordinates[reg_num]['EOVx'],
+                'null_point': self.data_handler.station_coordinates[reg_num]['null_point']
+            }
+
+        return stations
+
+    def create_rivers(self) -> dict:
+        """
+        Function for creating the rivers data structure. A river looks like
+        {'river_name': [station1, station2, ..., stationN]}, where the stations
+        inside the list are sorted in descending order by their null points.
+        :return dict: dictionary of all rivers
+        """
+        rivers_unsorted = self.get_rivers_unsorted()
+
+        return self.sort_rivers(rivers_unsorted=rivers_unsorted)
+
+    def get_rivers_unsorted(self) -> dict:
+        """
+        Creates dictionary of dictionaries like
+        {'river_name': [station1, station2, ..., stationN]}, where the stations
+        inside the list are not sorted.
+        :return dict: dictionary of unsorted rivers
+        """
+        rivers_unsorted = {}
+        for river_name in list(self.data_handler.river_station_mapping.keys()):
+            station_name_list = self.data_handler.river_station_mapping[river_name]
+            reg_number_list = [
+                self.data_handler.station_reg_mapping[station_name] for station_name in station_name_list
+            ]
+
+            rivers_unsorted[river_name] = reg_number_list
+
+        return rivers_unsorted
+
+    def sort_rivers(self, rivers_unsorted: dict) -> dict:
+        """
+        Sorts stations in rivers in descending order by their null points
+        :param dict rivers_unsorted: dictionary of unsorted rivers
+        :return dict: dictionary of sorted rivers
+        """
+        rivers_sorted = {}
+        for river_name in list(rivers_unsorted.keys()):
+            river_unsorted = rivers_unsorted[river_name]
+            river_sorted = sorted(
+                river_unsorted,
+                key=lambda x: -self.data_handler.station_coordinates[x]['null_point']
+            )
+
+            rivers_sorted[river_name] = river_sorted
+
+        return rivers_sorted
+
+    def create_completed_rivers(self) -> dict:
```

Comment:
Ez menthet közvetlenül a tagváltozóba, ha később nem akarjuk kívül is meghívni. Így csak a hívás lesz a `run` metódusban.

Ez a metódus menjen feljebb a fájlban, a többi `create_...` metódus utánra.